hrm-domain/hrm-service/src/profile/profile.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+import { TResult, isErr, newErr, newOk } from '@tech-matters/types';
+import {
+  Identifier,
+  Profile,
+  createIdentifierAndProfile,
+  getIdentifierWithProfile,
+} from './profile-data-access';
+export { Identifier, Profile };
+
+export const getOrCreateProfileWithIdentifier =
+  (task?) =>
+  async (
+    idx: string,
+    accountSid: string,
+  ): Promise<TResult<{ identifier: Identifier; profile: Profile }>> => {
+    try {
+      if (!idx) {
```

Comment:
Do we care about the get / create logic being atomic? At the moment, with no unique constraint being enforced there's a small chance we could duplicate identifiers here